datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -712,8 +728,12 @@ private DatastreamTask getDatastreamTask(String taskName) {
           err += " Queuing up a new onAssignmentChange event for retry.";
           _eventQueue.put(CoordinatorEvent.createHandleInstanceErrorEvent(ExceptionUtils.getRootCauseMessage(ex)));
           if (isDatastreamUpdate) {
+            _metrics.updateKeyedMeter(CoordinatorMetrics.getKeyedMeter(
+                CoordinatorEvent.EventType.HANDLE_DATASTREAM_CHANGE_WITH_UPDATE), 1);
```

Comment:
This specific metric is taken care in the generic exception catch of `handleEvents`. If you see value in the number of retries, then you will have to justify the reason and add another metric.

Reply:
Based on your earlier comments. I think the value is in knowing the number of tasks that failed to be processed. But probably it will help to have some additional context as well like which datastream or connector the task belonged to?